sycl/source/detail/kernel_impl.hpp

```diff
@@ -251,6 +255,17 @@ class kernel_impl {
   std::mutex *MCacheMutex = nullptr;
   mutable std::string MName;
 
+  using DeviceKernelInfoDeleterTy = void (*)(DeviceKernelInfo *);
+  using DeviceKernelInfoUniquePtr =
+      std::unique_ptr<DeviceKernelInfo, DeviceKernelInfoDeleterTy>;
+  DeviceKernelInfoUniquePtr MDeviceKernelInfo;
```

Comment:
I'd rather have 

```cpp
DeviceKernelInfo InteropInfoHolder;
DeviceKernelInfo &Info; // = IsInterop ? InteropInfoHolder : *PM.getInfo();
```

It will both be easier to read and avoid extra heap allocation.